enterprise/server/cmd/goinit/main.go

```diff
@@ -54,27 +66,19 @@ func chroot(path string) error {
 	return os.NewSyscallError("CHROOT", syscall.Chroot(path))
 }
 
-func reapChildren(ctx context.Context) {
+func reapChildren(ctx context.Context, reapMutex *sync.RWMutex) {
+	c := make(chan os.Signal, 1)
+	signal.Notify(c, unix.SIGCHLD)
+
 	for {
 		select {
 		case <-ctx.Done():
-			break
-		default:
-			// borrowed from https://github.com/driusan/dainit/blob/master/zombie.go
-
-			// If there are multiple zombies, Wait4 will reap one, and then block until the next child changes state.
-			// We call it with NOHANG a few times to clear up any backlog, and then make a blocking call until our
-			// next child dies.
+			return
+		case <-c:
+			reapMutex.Lock()
+			defer reapMutex.Unlock()
```

Comment:
Will this loop body ever execute more than once? (Seems like this might deadlock)